### PR TITLE
Refactor groups, interests, and locations

### PR DESCRIPTION
**Pull request by chalo2000**
## Overview

Redid interests to have CRUD-like endpoints like groups and locations, updated models based on API spec, and refactored the code to be stylistically consistent and easier.

## Changes Made

- Mimic @noah-solomon's general structure for groups and locations for interests
- Updated hard coded data to allow for subtitles and image URLs
- Moved `modify_attribute` to utils
- Simplified comments (since CRUD is easy to understand and any other endpoints are currently documented in the Notion API spec which should be moved in the future to a better documentation source)
- Explicitly pass status codes to remove ambiguity

## Test Coverage

Updated Postman suite (and next to come - Django tests #3 )

## Next Steps

#3 

  
  

</details>

### CODE REVIEW

File: src/person/controllers/authenticate_controller.py [hunk 1]
```diff
@@ -68,7 +68,9 @@ def process(self):
             token = self._data.get("id_token")
             token_info = self._get_token_info(token)
             if token_info is None:
-                return failure_response("ID Token is not valid.")
+                return failure_response(
+                    "ID Token is not valid.", status.HTTP_404_NOT_FOUND
```
Comment (noah-solomon) [inline src/person/controllers/authenticate_controller.py:72]: shouldn't this be 401 unauthorized

**Review by noah-solomon — APPROVED** [review]
looks good king

File: src/group/controllers/update_group_controller.py [hunk 2]
```diff
@@ -1,31 +1,35 @@
...
-        group = Group.objects.filter(id=self._group_id)
-        if not group:
-            return failure_response("Group does not exist")
-        if name is not None and group[0].name != name:
-            group[0].name = name
-        group[0].save()
-        return success_response(self._serializer(group[0]).data, status.HTTP_200_OK)
+        """Process a request to update a group's fields that have changed."""
+
+        # Get the model
```
Comment (alanna-zhou) [inline src/group/controllers/update_group_controller.py:17]: this is super great work, but these comments are a bit redundant i feel

File: src/interest/controllers/update_interest_controller.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+class UpdateInterestController:
+    def __init__(self, id, data, serializer):
+        self._id = id
+        self._data = data
+        self._serializer = serializer
+
+    def process(self):
+        """Process a request to update an interest's fields that have changed."""
+
```
Comment (alanna-zhou) [inline src/interest/controllers/update_interest_controller.py:16]: i rly would recommend deleting these coments haha

**Review by alanna-zhou — APPROVED** [review]
very cleannn 🧼

**Pull request merged**